icu4c/source/i18n/number_decimalquantity.cpp

```diff
@@ -1082,7 +1082,7 @@ UnicodeString DecimalQuantity::toScientificString() const {
     result.append(u'E');
     int32_t _scale = upperPos + scale + exponent;
     if (_scale == INT32_MIN) {
-        result.append({u"-2147483648", -1});
+        result.append(UnicodeString(true, u"-2147483648", -1));
```

Comment:
FYI
Adding overloads can still break certain call sites, but I think this is rare.
In this case, the old code also copied the contents; now I am aliasing it.